internal/parse/parser.go

```diff
@@ -273,11 +302,152 @@ func (p *Parser) parseGoFullName() (FullName, bool, error) {
 		return FullName{}, false,
 			fmt.Errorf("go object near char %d not qualified", p.pos)
 	}
+
 	cp.restore()
 	return FullName{}, false, nil
 }
 
-// parseInputExpression parses an input expression of the form $Type.name.
+// parseList takes a parsing function that returns a FullName and parses a
+// bracketed, comma seperated, list.
+func (p *Parser) parseList(parseFn func(p *Parser) (FullName, bool, error)) ([]FullName, bool, error) {
+	cp := p.save()
+
+	if p.skipByte('(') {
+		parenPos := p.pos
+		p.skipSpaces()
+
+		nextItem := true
+		var objs []FullName
+		for nextItem {
+			if obj, ok, err := parseFn(p); ok {
+				objs = append(objs, obj)
+			} else if err != nil {
+				return nil, false, err
+			} else {
+				return nil, false, fmt.Errorf("invalid identifier near char %d", p.pos)
+			}
+
+			p.skipSpaces()
+			if p.skipByte(')') {
+				return objs, true, nil
+			}
+
+			nextItem = p.skipByte(',')
+			p.skipSpaces()
+		}
+		return nil, false, fmt.Errorf("missing closing parentheses for char %d", parenPos)
+	}
+
+	cp.restore()
+	return nil, false, nil
+}
+
+// parseColumns parses a list of columns. For lists of more than one column the
+// columns must be enclosed in brackets e.g. "(col1, col2) AS &Person.*".
+func (p *Parser) parseColumns() ([]FullName, bool) {
+	cp := p.save()
+
+	// We skip a space here to keep consistent with parseTargets which also
+	// consumes one space before the start of the expression.
+	p.skipByte(' ')
+
+	// Case 1: A single column e.g. "p.name".
+	if col, ok, _ := p.parseColumn(); ok {
+		return []FullName{col}, true
+	}
+
+	// Case 2: Multiple columns e.g. "(p.name, p.id)".
+	if cols, ok, _ := p.parseList((*Parser).parseColumn); ok {
+		return cols, true
+	}
+
+	cp.restore()
+	return nil, false
+}
+
+// parseTargets parses the part of the output expression following the
+// ampersand. This can be one or more references Go objects.
+// If the ampersand is not preceded by a space and succeeded by a name or
+// opening bracket the targets are not parsed.
+func (p *Parser) parseTargets() ([]FullName, bool, error) {
+	cp := p.save()
+
+	// An '&' must be preceded by a space and succeeded by a name or opening
+	// bracket.
+	if p.skipString(" &") {
+
+		// Case 1: A single target e.g. "&Person.name".
+		if target, ok, err := p.parseGoFullName(); ok {
+			return []FullName{target}, true, nil
+		} else if err != nil {
+			return nil, false, err
+		}
+
+		// Case 2: Multiple targets e.g. "&(Person.name, Person.id)".
```

Comment:
I believe the agreement about the syntax was:
```
(&Person.name, &Person.id)
```

Reply:
I've changed the parser to recognise this. It's working all fine so it seems good to switch to this syntax.